.github/workflows/builds.yml

```diff
@@ -136,88 +177,151 @@ jobs:
       TERM: xterm-256color
 
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v3
-
-      # Runs a set of commands using the runners shell
       - name: Setup the image
         run: |
           brew install qt5
           brew install ninja
+
+      - uses: actions/checkout@v3
+        with:
+          path: input-leap
+          submodules: recursive
+
+      - name: Setup the build
+        run: |
+          . input-leap/macos_environment.sh
+          cmake -DCMAKE_BUILD_TYPE="${B_BUILD_TYPE}" -S input-leap -B build -G Ninja \
+                -DCMAKE_OSX_SYSROOT=$(xcode-select --print-path)/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk \
+                -DCMAKE_OSX_DEPLOYMENT_TARGET=10.9 -DCMAKE_UNITY_BUILD=1
+
       - name: Run the build
-        run: env B_CMAKE_FLAGS="-DCMAKE_UNITY_BUILD=1" sh -x ./clean_build.sh
+        run: |
+          cmake --build build --parallel
+
       - uses: actions/upload-artifact@v3
+        if: env.B_BUILD_TYPE == 'Release'
         with:
           name: Publish the DMG for ${{ matrix.os }}
           path: ./build/bundle/*.dmg
-          if-no-files-found: warn
-        if: ${{ env.B_BUILD_TYPE == 'Release' }}
+          if-no-files-found: error
 
-  win-build-2019:
-    runs-on: windows-2019
 
-    steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v2
-        with:
-          python-version: '3.7'
-          architecture: 'x64'
-      - name: Download Bonjour SDK-like
-        run: ./azure-pipelines/download_install_bonjour_sdk_like.ps1
```

Comment:
Yeah, this was something I raised before, Povilas. I don't quite understand your proposal, though.

Reply:
To me this moves the configuration from one CI (azure with scripts) to another CI (github yml). The [windows build instructions](https://github.com/input-leap/input-leap/wiki/Building-on-Windows) direct users to download Qt/Bonjour/etc from the respective sources, so I guess this comment only applies to the `clean_build.bat` not `download_install_bonjour_sdk_like` and `download_install_qt`?

Reply:
Yeah, and we could just do away with Azure, like, completely. I have no objection to duplicating build configs *now* that I've seen the implementation. It's clean, seems easy to maintain, and I don't think it's that big an issue. The azure-pipelines scripts aren't used by users as @Cimbali states, so that's not an issue.. right?

Reply:
I'm in favour of duplicate build configs. Let's not make the scripts anymore difficult to manage, and modifying the scripts will add complexity and increase maintenance.

Reply:
@p12tic Any further thoughts?

Reply:
`clean_build.sh` are not for users, but an easy way to build the project for developers. The fact that we're duplicating configs means that clean_build.sh is not right for the purpose, but we can change the script as well.

Reply:
@p12tic Sure, but users do use it too. I don't think duplicating configs is too big of an issue. I'd really like to get this PR merged, to be honest, and how often do we change the scripts or CI? AFAIK, not particularly often, which means maintainability isn't a massive concern.. right?

Reply:
OK, so before I push my rebase+squash, I just wanted to resolve this conversation, @p12tic.

I saw you approved the PR, but this review is outstanding. Another thing we should arrange is the cessation of Azure Pipelines - most people I see downloading nightlies _are_ using GH Actions.

We don't need two CIs. @Cimbali's work on this PR has been very well done, and we can cope with just one CI service. It will simplify things, and I have great confidence in the stability of this workflow.

Can we arrange for this PR to be merged? We have people asking about Windows installers, and macOS DMGs - and this will resolve that.

Reply:
@p12tic We need to make a decision. More cases of confusion r.e Azure and GH Actions. We need this review resolved so that GitHub will not complain when merging.

Reply:
I'm gonna go ahead and resolve this conversation. @p12tic, if you want to revisit the config in future, and duplication, we can definitely do that.

But as we've all approved the PR, I'm going to a) resolve this convo and b) merge the PR into master.

I hope that's OK.